isabelle/Doc/Specification/Core.thy

```diff
@@ -1,235 +1,12 @@
-start or end of the validity interval for the Marlowe transaction.\<close>
-
-subsection \<open>Observation\label{sec:observation}\<close>
-
-text \<open>An @{type Observation} evaluates to a logical value (true or false), given the current statet
-of the Marlowe contract.
-@{datatype [display,names_short, margin=40]Observation}
-\<close>
-
-text \<open>The @{term "ChoseSomething i"} term reports whether a choice @{term i} has been made thus far in
-the contract.\<close>
-
-text \<open>The terms @{term TrueObs} and @{term FalseObs} provide the logical constants @{value true} and
-@{value false}. The logical operators @{term "\<not> x"}, @{term "x \<and> y"},
-and @{term "x \<or> y"} are represented by the terms @{term "NotObs x"}, @{term "AndObs x y"}, and
-@{term "OrObs x y"}, respectively.\<close>
-
-text \<open>Value comparisons @{term "x < y"}, @{term "x \<le> y"}, @{text "x > y"}, @{text "x \<ge> y"},
-and @{term "x = y"} are represented by @{term "ValueLT x y"}, @{term "ValueLE x y"},
-@{term "ValueGT x y"}, @{term "ValueGE x y"}, and @{term "ValueEQ x y"}.\<close>
-
-subsection \<open>Token\label{sec:token}\<close>
-
-text \<open>Each currency in the contract is represented by a @{type Token} that consists of a symbol
-@{term CurrencySymbol} and a name @{term TokenName}, each represented by bytes.
-@{datatype [display,names_short, margin=40]Token}
-\<close> text \<open>FIXME: print type synonym: @{term [names_short, margin=40]CurrencySymbol}
-\<close> text \<open>FIXME: print type synonym: @{term [names_short, margin=40]TokenName}
-\<close>
-
-subsection \<open>State\label{sec:state}\<close>
-
-text \<open>The internal state of a Marlowe contract consists of the current balances in each party's
-account, a record of the most recent value of each type of choice, a record of the most recent
-value of each variable, and the minimum time for which input can be applied to the contract. The
-data for accounts, choices, and bound values are stored as association lists.\<close>
-(* Sadly there is no antiquote to print a record, and I wasn't able to
-make the snipet import work (described in chapter 7 of the Sugar Latex PDF).
-So to show records we need to duplicate the definition
- *)
-text \<open>FIXME: Print record:\<close>
-record State = accounts :: Accounts
-               choices :: "(ChoiceId \<times> ChosenNum) list"
-               boundValues :: "(ValueId \<times> int) list"
-               minTime :: POSIXTime
-text \<open>FIXME: print type synonym: @{term [names_short, margin=40]Accounts}\<close>
-
-subsection \<open>Environment\label{sec:environment}\<close>
-
-text \<open>The execution environment of a Marlowe contract simply consists of the (inclusive) time
-interval within which the transaction is occurring.\<close>
-record Environment = timeInterval :: TimeInterval
-text \<open>FIXME: Print record:
-@{term [names_short, margin=40]TimeInterval}
-\<close>
 
```

Comment:
> `text \<open>FIXME: Print record: @{term [names_short, margin=40]Transaction}`

We could eliminate this problem by moving this section into Core and making it literate there.